lib/discovery/README.md

```diff
@@ -0,0 +1,15 @@
+# Discovery
+
+> Discover other peers via the DHT or MDNS and replicate with them
+
+## Usage
+
+The `Discovery` class provides an abstraction layer that allows peer discovery using either a [distributed hash table](https://en.wikipedia.org/wiki/Distributed_hash_table) (DHT) or [Multicast DNS](https://en.wikipedia.org/wiki/Multicast_DNS) (mDNS).
+
```

Comment:
We could add a snippet like this:

```js
import Hypercore from 'hypercore'
import { KeyManager } from '@mapeo/crypto'
import { Discovery } from './index.js'

const rootKey = KeyManager.generateRootKey()
const keyManager = new KeyManager(rootKey)
const identityKeyPair = keyManager.getIdentityKeypair()

const core = new Hypercore()
await core.ready()

const discover = new Discovery({
	identityKeyPair,
	mdns: true,
	dht: true,
})

await discover.ready()

discover.on('connection', async (connection, peer) => {
	core.replicate(connection)
})

await discover.join(core.discoveryKey)
```

This is likely the simplest possible usage of the discovery module.

Reply:
Addressed via 179ba173c7c1d0f8212bb60e52d4a7b71508eb84